model/user/user.go

```diff
@@ -166,6 +168,53 @@ func (u *DBUser) UpdateParsleySettings(settings parsley.Settings) error {
 	return nil
 }
 
+// CheckAndUpdateSchedulingLimit checks if the number of tasks to be activated by the user is allowed given
+// the global per-user hourly task scheduling limit, and updates relevant timestamp and counter info used
+// to track the user's hourly scheduling usage. The numTasksActivated parameter being negative signifies
+// the user is deactivating tasks, which frees up space in their scheduling limit.
+func (u *DBUser) CheckAndUpdateSchedulingLimit(settings *evergreen.Settings, numTasksModified int) error {
+	var update bson.M
+	maxScheduledTasks := settings.TaskLimits.MaxHourlyPatchTasks
+	if maxScheduledTasks == 0 {
+		return nil
+	}
+	if numTasksModified > maxScheduledTasks {
+		return errors.Errorf("cannot schedule %d tasks, maxumum hourly per-user limit is %d", numTasksModified, maxScheduledTasks)
+	}
+	now := time.Now()
+	oneHourAgo := now.Add(-1 * time.Hour)
+	// If the last time the user scheduled patch tasks was within the hour, increment the number
+	// of activated tasks to the user's counter, erroring if the global limit is breached. If numTasksModified
+	// is negative, the counter is decremented.
+	if u.LastScheduledTasksAt.After(oneHourAgo) {
+		update = bson.M{
+			"$set": bson.M{NumScheduledPatchTasksKey: getNewCounter(u.NumScheduledPatchTasks, numTasksModified)},
+		}
```

Comment:
It seems like this means we won't allow users to unschedule tasks if they're above their quota, unless they schedule enough to go below it

Reply:
This should now be covered by the new activated param. Added a test case